storage/client/client.go

```diff
@@ -1121,6 +1121,179 @@ func (c *StorageClient) DebondingDelegationsTo(ctx context.Context, address stak
 	return &ds, nil
 }
 
+// StakingRewards returns a list of consensus staking rewards for a given account.
+func (c *StorageClient) StakingRewards(ctx context.Context, address staking.Address, year uint64, p apiTypes.GetConsensusAccountsAddressStakingRewardsYearParams) (*StakingRewardList, error) {
+	// Validate year, find starting snapshot of account's delegations at year start
+	// Fetch all escrow events where delegator==address from that year, this includes staking reward disbursements which have newShares == 0
```

Comment:
Seems like a lot of work needs to be done on this query. Could we compute this once for each past year and every account and and store in the DB? Via some background job? And then for the most recent year we can compute it daily and update the results.